src/notation/internal/notationparts.cpp

```diff
@@ -626,6 +626,10 @@ void NotationParts::replaceDrumset(const InstrumentKey& instrumentKey, const Dru
 
     score()->undo(new mu::engraving::ChangeDrumset(instrument, &newDrumset));
 
+    if (part->staves().size() > 0) {
+        score()->setLayout(Fraction(0, 1), score()->endTick(), part->staves().front()->idx(), part->staves().back()->idx());
```

Comment:
Wouldn't this need to be done inside the engraving module, so that it is also triggered on undo/redo?